homeassistant/components/iaqualink/climate.py

```diff
@@ -103,7 +103,14 @@ def max_temp(self) -> int:
     @property
     def target_temperature(self) -> float:
         """Return the current target temperature."""
-        return float(self.dev.state)
+        if not isinstance(self.dev.state, str) or self.dev.state == "":
```

Comment:
What types have we encountered here?

Reply:
Well, I am just a user of the integration and like the capabilities. My understanding is that this is a "scrape" style integration so I'm sure it's difficult to keep up with anything close to an API.

My logs only show blanks in the exceptions.

Reply:
The library doesn't do scraping, it does use an API.

I don't think target_temperature should be anything but a string or be empty. Would you mind submitting redacted debug logs to show what devices are being returned?